### PR TITLE
FrenchRidgeLap

### PR DESCRIPTION
**Pull request by papachap**
image

Implementation of the `FrenchRidgeLap` BTLx process and `LFrenchRidgeLapJoint`. 
In this PR old modules concerning the FRL are being removed as well. 

**After this PR we can proceed and do a major cleanup of the repo by updating and removing stuff that belonged to the old system of implementation.**

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [x] I added new functions/classes and made them available on a second-level import, e.g. `compas_timber.datastructures.Beam`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_timber/connections/l_french_ridge_lap.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            raise BeamJoinningError(self.elements, self, debug_info=str(ae), debug_geometries=geometries)
+        except Exception as ex:
+            raise BeamJoinningError(self.elements, self, debug_info=str(ex))
+        self.beam_a.add_blank_extension(start_a, end_a, self.guid)
+        self.beam_b.add_blank_extension(start_b, end_b, self.guid)
+
+    def add_features(self):
+        """Adds the required extension and trimming features to both beams.
+
+        This method is automatically called when joint is created by the call to `Joint.create()`.
```
Comment (obucklin) [inline src/compas_timber/connections/l_french_ridge_lap.py:128]: isn't this called on `beam.geometry`, not `Joint.create()`?

File: src/compas_timber/connections/l_french_ridge_lap.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    def cutting_plane_b(self):
+        # the plane that cuts beam_a
+        ref_side_dict = beam_ref_side_incidence(self.beam_a, self.beam_b, ignore_ends=True)
+        ref_side_index = max(ref_side_dict, key=ref_side_dict.get)
+        return self.beam_b.ref_sides[ref_side_index]
+
+    def add_extensions(self):
+        """Calculates and adds the necessary extensions to the beams.
+
+        This method is automatically called when joint is created by the call to `Joint.create()`.
```
Comment (obucklin) [inline src/compas_timber/connections/l_french_ridge_lap.py:103]: This is called in `Model.process_joinery()`

File: src/compas_timber/connections/l_french_ridge_lap.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+            self.beam_a, self.beam_b, self.cutting_plane_b, self.drillhole_diam, self.beam_a_ref_side_index
+        )
+        frl_b = FrenchRidgeLap.from_beam_beam_and_plane(
+            self.beam_b, self.beam_a, self.cutting_plane_a, self.drillhole_diam, self.beam_b_ref_side_index
+        )
+        self.beam_a.add_features(frl_a)
+        self.beam_b.add_features(frl_b)
+        self.features = [frl_a, frl_b]
+
+    def check_geometry(self):
```
Comment (obucklin) [inline src/compas_timber/connections/l_french_ridge_lap.py:149]: We should consider giving this a standardized name

**Review by obucklin — COMMENTED** [review]
in general looks fine. fix the few docstrings i commented on and let's powwow for 100 seconds with @chenkasirer about the name

**Review by papachap — COMMENTED** [review]
Ready for re-review. I pulled the latest changes from main and made changes based on the feedback.

**Comment by obucklin** [comment]
LGTM

**Review by obucklin — APPROVED** [review]

**Pull request merged**